File: 4-How_to_solve_coding_problems/contains_common_item.py
def contains_common_item(lst1, lst2):
    # Loop through first list and create object where properties === items in
    # the array
    dict = {}

    for i in lst1:
        if i not in dict:
            dict[i] = True

    # loop through second list and check if item in second list exists on
    # created object
    for j in lst2:
        if j in dict:
            return True

    return False


# O(a + b) Time Complexity
list1 = ["a", "b", "c", "x"]
list2 = ["z", "y", "a"]

result = contains_common_item(list1, list2)
print(result)


# def contains_common_item2(lst1, lst2):
#     if set(lst1) & set(lst2):
#         return True

#     return False


# result = contains_common_item2(list1, list2)
# print(result)

File: 4-How_to_solve_coding_problems/contains_common_item.js
function containsCommonItem(arr1, arr2) {
  // Loop through first array and create object where properties === items in the
  // array
  let map = {};

  for (let i = 0; i < arr1.length; ++i) {
    if (!map[arr1[i]]) {
      const item = arr1[i];
      map[item] = true;
    }
  }

  // loop through second array and check if item in second array exists on created
  // object
  for (let j = 0; j < arr2.length; ++j) {
    if (map[arr2[j]]) {
      return true;
    }
  }

  return false;
}

// O(a + b) Time Complexity
const array1 = ['a', 'b', 'c', 'x'];
const array2 = ['z', 'y', 'a'];

const result = containsCommonItem(array1, array2);
console.log(result);

// function containsCommonItem2(arr1, arr2) {
//   return arr1.some(item => arr2.include(item));
// }
//
// const result = containsCommonItem(array1, array2);
// console.log(result);
